tools/update-npm.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -o pipefail -o errexit
+
+# This script will detect if we're running with an npm that
+# does not support CI, and will upgrade it if so
+MINIMUM_VERSION=5.7.0
+CURRENT_VERSION=$(npm -v)
+
+echo "Ensuring npm version is >= ${MINIMUM_VERSION}"
+
+# Sort the versions using `sort`s version sort, and take the
+# first line
+NEWEST_VERSION="$(echo -e "${CURRENT_VERSION}\n${MINIMUM_VERSION}" | sort -V -r | head -n1)"
+
+# If the head is the same as the current, we are already >=
+# the minimum version
+if [ "${CURRENT_VERSION}" = "${NEWEST_VERSION}" ]; then
+	echo 'Npm already supports npm ci, not upgrading';
+else
+	echo 'Upgrading npm version to support npm ci';
+	npm install -g npm@${MINIMUM_VERSION}
```

Comment:
So, to my understansing, we can have different base images with different node/npm versions.

Why not updating base images to make sure they do have a minimal npm version we need.
So, in i386-nlp image, we know that npm version is low, so we add a command there to upgrade it.

Reply:
https://github.com/balena-io/balena-supervisor/pull/1017#discussion_r296667485